cumulusci/core/keychain/BaseEncryptedProjectKeychain.py

```diff
@@ -46,6 +46,8 @@ def _set_encrypted_service(self, service, encrypted, project):
         self.services[service] = encrypted
 
     def _set_org(self, org_config, global_org):
+        org_config.keychain = self
+        org_config.global_org = global_org
```

Comment:
Perhaps we should have some sanity-check assertions to make sure that these are not already set to different values.